platforms/forge/src/main/java/com/dfsek/terra/forge/AwfulForgeHacks.java

```diff
@@ -53,38 +70,83 @@ public static JarFile getTerraJar() throws IOException {
                                     .equals(ForgeEntryPoint.class.getName().replace('.', '/') + ".class")))
                     .findFirst()
                     .orElseThrow(() -> new IllegalStateException("Could not find Terra JAR"));
-        
     }
     
     public static void loadAllTerraClasses() {
-        
-        try(JarFile jar = getTerraJar()) {
-            jar.stream()
-               .forEach(jarEntry -> {
-                   if(jarEntry.getName().startsWith("com/dfsek/terra/forge/mixin")
-                   || jarEntry.getName().startsWith("com/dfsek/terra/mod/mixin")) {
-                       return;
-                   }
-                   if(jarEntry.getName().endsWith(".class")) {
-                       String name = jarEntry.getName().replace('/', '.');
-                       name = name.substring(0, name.length() - 6);
-                       try {
-                           Class.forName(name);
-                       } catch(ClassNotFoundException | NoClassDefFoundError e) {
-                           LOGGER.warn("Failed to load class {}: {}", name, e);
+        if (FMLLoader.isProduction()) {
+            try(JarFile jar = getTerraJar()) {
+                jar.stream()
+                   .forEach(jarEntry -> {
+                       if(jarEntry.getName().startsWith("com/dfsek/terra/forge/mixin")
+                          || jarEntry.getName().startsWith("com/dfsek/terra/mod/mixin")) {
+                           return;
+                       }
+                       if(jarEntry.getName().endsWith(".class")) {
+                           String name = jarEntry.getName().replace('/', '.');
+                           name = name.substring(0, name.length() - 6);
+                           try {
+                               Class.forName(name);
+                           } catch(ClassNotFoundException | NoClassDefFoundError e) {
+                               LOGGER.warn("Failed to load class {}: {}", name, e);
+                           }
                        }
-                   }
-               });
-        } catch(IOException e) {
-            throw new IllegalStateException("Could not load all Terra classes", e);
+                   });
+            } catch(IOException e) {
+                throw new IllegalStateException("Could not load all Terra classes", e);
+            }
+        } else {
+            // Forgive me for what I'm about to do...
```

Comment:
This should not be in production code at all. Either put it in a completely separate source set, or remove it.

Reply:
I don't see why it's an issue. As we've agreed I have the burden of maintaining the forge version I can tell you that the ability to run in a dev env is a pretty necessary thing for me. The code is only run if in a dev env and even if it somehow gets run instead of the normal awful hack it will just class not found. If it's moved into a different source set it will also class not found, there is practically no change to the end user. Moving it to another source set will just clutter up the source code. And it's not like the forge impl is the peak of correctness this is literally in `AwfulForgeHacks`. But if your deadset on it being in a different source set I will move it

Reply:
I still dont like that its even there at all, but i guess there is no other option.